scripts/download-rc.js

```diff
@@ -0,0 +1,161 @@
+        } else {
+            console.log(`Downloading: hazelcast enterprise jar `
+                + `com.hazelcast:hazelcast-enterprise:${HAZELCAST_ENTERPRISE_VERSION}`);
+            const subprocess = spawnSync('mvn', [
+                '-q',
+                'dependency:get',
+                `-DrepoUrl=${ENTERPRISE_REPO}`,
+                `-Dartifact=com.hazelcast:hazelcast-enterprise:${HAZELCAST_ENTERPRISE_VERSION}`,
+                `-Ddest=hazelcast-enterprise-${HAZELCAST_ENTERPRISE_VERSION}.jar`
+            ], {
+                stdio: 'inherit',
+                shell: ON_WINDOWS
+            });
+            if (subprocess.status !== 0) {
+                throw `Failed download hazelcast enterprise jar `
+                + `com.hazelcast:hazelcast-enterprise:${HAZELCAST_ENTERPRISE_VERSION}`;
+            }
+        }
+
+        if (fs.existsSync(`hazelcast-enterprise-${HAZELCAST_TEST_VERSION}-tests.jar`)) {
+            console.log('hazelcast-enterprise-tests.jar already exists, not downloading from maven.');
+        } else {
+            console.log('Downloading: hazelcast enterprise test jar '
+                + `com.hazelcast:hazelcast-enterprise:${HAZELCAST_TEST_VERSION}:jar:tests`);
+            const subprocess = spawnSync('mvn', [
+                '-q',
+                'org.apache.maven.plugins:maven-dependency-plugin:2.8:get',
+                `-DrepoUrl=${ENTERPRISE_TEST_REPO}`,
+                `-Dartifact=com.hazelcast:hazelcast-enterprise:${HAZELCAST_TEST_VERSION}:jar:tests`,
+                `-Ddest=hazelcast-enterprise-${HAZELCAST_TEST_VERSION}-tests.jar`
+            ], {
+                stdio: 'inherit',
+                shell: ON_WINDOWS
+            });
+            if (subprocess.status !== 0) {
+                throw 'Failed to download hazelcast enterprise test jar '
+                + `com.hazelcast:hazelcast-enterprise:${HAZELCAST_TEST_VERSION}:jar:tests`;
+            }
+        }
+        console.log('Starting Remote Controller ... enterprise ...');
+    } else {
+        if (fs.existsSync(`hazelcast-${HAZELCAST_VERSION}.jar`)) {
+            console.log('hazelcast.jar already exists, not downloading from maven.');
+        } else {
+            console.log(`Downloading: hazelcast jar com.hazelcast:hazelcast:${HAZELCAST_VERSION}`);
+            const subprocess = spawnSync('mvn', [
+                '-q',
+                'dependency:get',
+                `-DrepoUrl=${REPO}`,
+                `-Dartifact=com.hazelcast:hazelcast:${HAZELCAST_VERSION}`,
+                `-Ddest=hazelcast-${HAZELCAST_VERSION}.jar`
+            ], {
+                stdio: 'inherit',
+                shell: ON_WINDOWS
+            });
+            if (subprocess.status !== 0) {
+                throw `Failed download hazelcast jar com.hazelcast:hazelcast:${HAZELCAST_VERSION}`;
+            }
+        }
+    }
+};
+
+// download the remote controller when the script is invoked directly
+if (require.main === module) {
```

Comment:
Do we need to call this script directly somewhere?

Reply:
We could. What if if you only want to download rc files

Reply:
I don't see where we need this functionality ATM. If we download RC, we run it immediately.